planner/core/encode.go

```diff
@@ -57,8 +58,19 @@ func (pn *planEncoder) encodePlanTree(p PhysicalPlan) string {
 }
 
 func (pn *planEncoder) encodePlan(p PhysicalPlan, isRoot bool, depth int) {
+	var storeType kv.StoreType = kv.UnSpecified
+	if !isRoot {
+		switch copPlan := p.(type) {
+		case *PhysicalTableReader:
+			storeType = copPlan.StoreType
+		case *PhysicalTableScan:
+			storeType = copPlan.StoreType
+		default:
+			storeType = kv.TiKV
+		}
+	}
 	actRows, analyzeInfo, memoryInfo, diskInfo := getRuntimeInfo(p.SCtx(), p)
-	plancodec.EncodePlanNode(depth, p.ID(), p.TP(), isRoot, p.statsInfo().RowCount, p.ExplainInfo(), actRows, analyzeInfo, memoryInfo, diskInfo, &pn.buf)
+	plancodec.EncodePlanNode(depth, p.ID(), p.TP(), isRoot, storeType, p.statsInfo().RowCount, p.ExplainInfo(), actRows, analyzeInfo, memoryInfo, diskInfo, &pn.buf)
```

Comment:
I think we can pass the encoded string of task deriectly.

Reply:
Comment addressed. PTAL~ @hanfei1991